### ISSUE TITLE
Support unqualified column lookups in parser

### ISSUE THREAD

**Issue by lukaseder** [jOOQ/jOOQ#10075]
We want to be able to look up unqualified column names and column names from aliased tables in the parser, such as:

```sql
-- These resolve to t.col
SELECT col FROM t 
SELECT (SELECT col) FROM t

-- This throws an ambiguous column exception, if both t.col and u.col exist
SELECT col FROM t, u

-- This resolves to u.col
SELECT col FROM t AS u
```

A subtask of #9061.

**Comment by lukaseder** [comment]
Fixed for jOOQ 3.14.0. Backport scheduled for 3.13.2 (#10076)

**Comment by lukaseder** [comment]
Fix commited via https://github.com/jOOQ/jOOQ/commit/1a15bfbf29cd8765bffd3b240da1bec1a3641b07 (#9061)
